File: Iniciante/1021 - Notas e Moedas/C/1021 - Notas e Moedas.c
#include <stdio.h>
#include <math.h> // biblioteca matemática

int main() {
	double valor, resto;
	
	scanf("%lf", &valor); // entrada
	
	// atribuição de valores das notas
	int n100 = floor(valor/100);
	resto = fmod(valor, 100);
	
	int n50 = floor(resto/50);
	resto = fmod(resto, 50);
	
	int n20 = floor(resto/20);
	resto = fmod(resto, 20);
	
	int n10 = floor(resto/10);
	resto = fmod(resto, 10);
	
	int n5 = floor(resto/5);
	resto = fmod(resto, 5);
	
	int n2 = floor(resto/2);
	resto = fmod(resto, 2);
	
	// atribuição de valores das moedas
	int n1 = floor(resto/1);
	resto = fmod(resto, 1);
	
	int n050 = floor(resto/0.50);
	resto = fmod(resto, 0.50);
	
	int n025 = floor(resto/0.25);
	resto = fmod(resto, 0.25);
	
	int n010 = floor(resto/0.10);
	resto = fmod(resto, 0.10);
	
	int n005 = floor(resto/0.05);
	resto = fmod(resto, 0.05);
	
	int n001 = floor(resto/0.01);

	// saída
	printf("NOTAS:\n");
	printf("%d nota(s) de R$ 100.00\n", n100);
	printf("%d nota(s) de R$ 50.00\n", n50);
	printf("%d nota(s) de R$ 20.00\n", n20);
	printf("%d nota(s) de R$ 10.00\n", n10);
	printf("%d nota(s) de R$ 5.00\n", n5);
	printf("%d nota(s) de R$ 2.00\n", n2);
	
	printf("MOEDAS:\n");
	printf("%d moeda(s) de R$ 1.00\n", n1);
	printf("%d moeda(s) de R$ 0.50\n", n050);
	printf("%d moeda(s) de R$ 0.25\n", n025);
	printf("%d moeda(s) de R$ 0.10\n", n010);
	printf("%d moeda(s) de R$ 0.05\n", n005);
	printf("%d moeda(s) de R$ 0.01\n", n001);
	
	return 0;
}

File: Iniciante/1021 - Notas e Moedas/Python/1021 - Notas e Moedas.py
# entrada do valor monetário
N = float(input())

if N > 0 and N < 1000000.00:
    # decomposição das notas
    n100 = int(N/100)
    resto = N%100

    n50 = int(resto/50)
    resto = resto%50

    n20 = int(resto/20)
    resto = resto%20

    n10 = int(resto/10)
    resto = resto%10

    n5 = int(resto/5)
    resto = resto%5

    n2 = int(resto/2)
    resto = resto%2

    # decomposição das moedas
    n1 = int(resto/1)
    resto = resto%1

    n050 = int(resto/0.50)
    resto = resto%0.50

    n025 = int(resto/0.25)
    resto = resto%0.25

    n010 = int(resto/0.10)
    resto = resto%0.10

    n005 = int(resto/0.05)
    resto = resto%0.05

    n001 = int(resto/0.01)

    # saída do valor formatado
    print('NOTAS:')
    print(n100, 'nota(s) de R$ 100.00')
    print(n50, 'nota(s) de R$ 50.00')
    print(n20, 'nota(s) de R$ 20.00')
    print(n10, 'nota(s) de R$ 10.00')
    print(n5, 'nota(s) de R$ 5.00')
    print(n2, 'nota(s) de R$ 2.00')

    print('MOEDAS:')
    print(n1, 'moeda(s) de R$ 1.00')
    print(n050, 'moeda(s) de R$ 0.50')
    print(n025, 'moeda(s) de R$ 0.25')
    print(n010, 'moeda(s) de R$ 0.10')
    print(n005, 'moeda(s) de R$ 0.05')
    print(n001, 'moeda(s) de R$ 0.01')